core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -959,6 +959,16 @@ public void testLoadMissingTable() {
         .hasMessageStartingWith("Table does not exist: ns.tbl");
   }
 
+  @Test
+  public void testLoadTableWithMissingMetadatafile() {
+    C catalog = catalog();
+
+    assertThat(catalog.tableExists(TBL)).as("Table should not exist").isFalse();
+    assertThatThrownBy(() -> catalog.loadTable(TBL))
+        .isInstanceOf(NoSuchTableException.class)
```

Comment:
the test doesn't reproduce the issue you're trying to repro and it passes without applying the fix that is being proposed

Reply:
Fixed it.